core/src/dird/ua_acl.cc

```diff
@@ -205,25 +174,27 @@ bool UaContext::AclAccessOk(int acl,
 
   retval = FindInAclList(user_acl->ACL_lists[acl], acl, item, len);
 
-  /*
-   * If we didn't find a matching ACL try to use the profiles this console is
-   * connected to.
-   */
-  if (!retval && user_acl->profiles && user_acl->profiles->size()) {
-    ProfileResource* profile = nullptr;
+  /* If we didn't find a matching ACL try to use the profiles this console is
+   * connected to. */
+  if (!retval.has_value()) {
+    if (user_acl->profiles && user_acl->profiles->size()) {
+      ProfileResource* profile = nullptr;
 
-    foreach_alist (profile, user_acl->profiles) {
-      retval = FindInAclList(profile->ACL_lists[acl], acl, item, len);
+      foreach_alist (profile, user_acl->profiles) {
+        retval = FindInAclList(profile->ACL_lists[acl], acl, item, len);
 
-      // If we found a match break the loop.
-      if (retval) { break; }
+        // If we found a match break the loop.
+        if (retval.has_value()) { break; }
+      }
```

Comment:
This looks a bit nondeterministic.  E.g. if im connected to two profiles and one of them disallows `restore` while the other one allows it, then its up to chance if i get to restore or not.

Do you think we should fix this ? For example we could only allow a command if either *any* profile allows it or if *all* profiles allow it.
I would prefer the *any* case, but since we know use three boolean states, *all* would also make some sense.